.github/workflows/docs.yaml

```diff
@@ -1,5 +1,7 @@
 name: Build docs
 on: [push, pull_request]
+env:
+  SPHINXOPTS: '-W'
```

Comment:
Should we move this to `doc/Makefile` so that local doc builds are the same of the CI ones?

Reply:
I definitely don't love putting logic in this file. For development purposes - it is pretty frustrating that it only shows one warning at a time. I wish it were implemented differently. For development purposes I think printing all the warnings is essential. I think it might be worth implementing this a different way than using -W. Hmm...

Reply:
I suppose that once this is merged, only new warnings introduced by someone's changes will fail the build. Not ideal but not too bad?

Reply:
Ummm... https://github.com/galaxyproject/galaxy/pull/11103/commits/9dd73c3590bab5e6948cd76282b55958e4d13009  maybe?

Reply:
Good for me!